lib/ace/transport_app.rb

```diff
@@ -214,44 +214,46 @@ def self.trusted_facts(certname)
         validate_schema(@schemas["run_task"], body)
         opts = body['target'].merge('protocol' => 'remote')
 
-        # This is a workaround for Bolt due to the way it expects to receive the target info
-        # see: https://github.com/puppetlabs/bolt/pull/915#discussion_r268280535
-        # Systems calling into ACE will need to determine the nodename/certname and pass this as `name`
-        target = [Bolt::Target.new(body['target']['host'] || body['target']['name'], opts)]
+        inventory = Bolt::Inventory.empty
+        target_data = {
+          'name' => body['target']['host'] || body['target']['name'] || 'remote',
```

Comment:
It seems like we used to be able to get away with a target with no name???? Not sure what the impact of this will be...

Reply:
This is in my mind the biggest outstanding question.